homeassistant/components/wiz_light/light.py

```diff
@@ -0,0 +1,368 @@
+                return features
+            except WizLightNotKnownBulb:
+                _LOGGER.info(
+                    "Bulb is not present in the library. Fallback to full feature"
+                )
+                return SUPPORT_FEATURES_RGB
+        # fallback
+        return SUPPORT_FEATURES_RGB
+
+    @property
+    def effect(self):
+        """Return the current effect."""
+        return self._effect
+
+    @property
+    def effect_list(self):
+        """Return the list of supported effects.
+
+        URL: https://docs.pro.wizconnected.com/#light-modes
+        """
+        return self._scenes
+
+    @property
+    def available(self):
+        """Return if light is available."""
+        return self._available
+
+    async def async_update(self):
+        """Fetch new state data for this light."""
+        await self.update_state()
+
+        if self._state is not None and self._state is not False:
+            self.update_brightness()
+            self.update_temperature()
+            self.update_color()
+            self.update_effect()
+
+    @property
+    def device_info(self):
+        """Get device specific attributes."""
+        return {
+            "connections": {(CONNECTION_NETWORK_MAC, self._mac)},
+            "name": self._attr_name,
+            "manufacturer": "WiZ Light Platform",
+            "model": f"{self._bulbtype.name} Mac: {self._mac}",
+        }
+
+    def update_state_available(self):
+        """Update the state if bulb is available."""
+        self._state = self._light.status
+        self._available = True
+
+    def update_state_unavailable(self):
+        """Update the state if bulb is unavailable."""
+        self._state = False
+        self._available = False
+
+    async def update_state(self):
+        """Update the state."""
+        try:
+            await self._light.updateState()
+            if self._light.state is None:
+                self.update_state_unavailable()
+            else:
+                self.update_state_available()
+        except (TimeoutError, WizLightTimeOutError) as ex:
+            _LOGGER.debug(ex)
+            self.update_state_unavailable()
+        _LOGGER.debug(
+            "[wizlight %s] updated state: %s and available: %s",
+            self._light.ip,
+            self._state,
+            self._available,
+        )
+
+    def update_brightness(self):
+        """Update the brightness."""
+        if self._light.state.get_brightness() is None:
+            return
+        try:
+            brightness = self._light.state.get_brightness()
+            if 0 <= int(brightness) <= 255:
+                self._brightness = int(brightness)
+            else:
+                _LOGGER.error(
+                    "Received invalid brightness : %s. Expected: 0-255", brightness
+                )
+                self._brightness = None
+        # pylint: disable=broad-except
+        except Exception as ex:
```

Comment:
What does this except guard against?

Reply:
I'm not sure anymore about this. Maybe a relic from older times. I have to check where a broad exception can occur.

Reply:
Let's take out the broad except handlers as it looks like they aren't needed anymore.  If they turn out to be needed, please catch only the expected exceptions